packages/next-auth/src/next/middleware.ts

```diff
@@ -103,16 +104,18 @@ async function handleMiddleware(
 ) {
   const { pathname, search, origin } = req.nextUrl
 
-  const signInPage = options?.pages?.signIn ?? "/api/auth/signin"
-  const errorPage = options?.pages?.error ?? "/api/auth/error"
+  const nextJsBasePath = getNextJsBasePath(req.nextUrl)
+  const fullPathname = nextJsBasePath + pathname
+  const signInPage = options?.pages?.signIn ?? `${nextJsBasePath}/api/auth/signin`
+  const errorPage = options?.pages?.error ?? `${nextJsBasePath}/api/auth/error`
   const basePath = parseUrl(process.env.NEXTAUTH_URL).path
```

Comment:
If your `NEXTAUTH_URL` contains a custom base path, for example, "https://example.com/custom-base", this line will extract the correct `basePath` as `/custom-base`, otherwise it will return `/api/auth` as default. Can't we use this to get the custom base path if provided instead of adding a new function? 🤔
I noticed that in your reproduction,  the `NEXTAUTH_URL` contains `/api/auth` which is incorrect.

Reply:
Hey, I configured the reproduction like it states in the documentation about the custom base path: https://next-auth.js.org/configuration/options#nextauth_url

Reply:
Oh sorry, I totally missed that!

Reply:
I updated the packages to the latest version and add a version with `NEXTAUTH_URL=http://localhost:3000/custom-base`.
https://github.com/psperber/next-auth-issues